src/pages/Swap/index.tsx

```diff
@@ -302,7 +301,7 @@ export function Swap({
   const fiatValueOutput = useUSDPrice(parsedAmounts[Field.OUTPUT])
 
   const [routeNotFound, routeIsLoading, routeIsSyncing] = useMemo(
-    () => [!trade?.swaps, TradeState.LOADING === tradeState, TradeState.SYNCING === tradeState],
+    () => [!trade?.swaps, TradeState.LOADING === tradeState, TradeState.LOADING === tradeState && Boolean(trade)],
```

Comment:
nit: everywhere else you check `tradeState === TradeState.XYZ`, here it is flipped.

Reply:
When I read this, I want to read "when the tradeState is LOADING". Everywhere else, you check `tradeState === TradeState.LOADING`, which follows my mental model. However, here, you check `TradeState.LOADING === tradeState`: "when LOADING is the tradeState".

In general, I like my conditionals to be "when VARIABLE is CONSTANT", and not the other way around. Can I lint for this?